static-analysis-plugin/plugin/src/main/groovy/com/novoda/staticanalysis/findbugs/FindbugsConfigurator.groovy

```diff
@@ -0,0 +1,111 @@
+package com.novoda.staticanalysis.findbugs
+
+import com.novoda.staticanalysis.StaticAnalysisExtension
+import com.novoda.staticanalysis.Violations
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.plugins.quality.FindBugs
+import org.gradle.api.plugins.quality.FindBugsExtension
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.JavaExec
+import org.gradle.internal.logging.ConsoleRenderer
+
+class FindbugsConfigurator {
+
+    void configure(Project project, Violations violations, StaticAnalysisExtension extension, Task evaluateViolations) {
+        extension.ext.findbugs = { Closure config ->
+            project.apply plugin: QuietFindbugsPlugin
+            List<String> excludes = []
+            configureExtension(project.extensions.findByType(FindBugsExtension), excludes, config)
+            project.afterEvaluate {
+                configureAndroidIfNeeded(project)
+                project.tasks.withType(FindBugs) { FindBugs findBugs ->
+                    configureTask(project, findBugs, evaluateViolations, violations, excludes)
+                }
+            }
+        }
+    }
+
+    private void configureAndroidIfNeeded(Project project) {
+        boolean isAndroidApp = project.plugins.hasPlugin('com.android.application')
+        boolean isAndroidLib = project.plugins.hasPlugin('com.android.library')
+        if (isAndroidApp || isAndroidLib) {
+            def variants = isAndroidApp ? project.android.applicationVariants : project.android.libraryVariants
+            configureAndroid(project, variants)
+        }
+    }
```

Comment:
yeah, I was following the [_rule of three_](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) to avoid to over-abstract (again) too early. A follow-up PR(s) can help tidy up and smooth out APIs